libs/utils/src/backoff.rs

```diff
@@ -1,11 +1,15 @@
 use std::fmt::{Debug, Display};
 
 use futures::Future;
+use rand::Rng;
 use tokio_util::sync::CancellationToken;
 
 pub const DEFAULT_BASE_BACKOFF_SECONDS: f64 = 0.1;
```

Comment:
In this PR we are introducing a 'network' timeout and a default one... but isn't the default one also mostly (exclusively?) used for network operations?